torch/__init__.py

```diff
@@ -1442,10 +1442,13 @@ def fn(model: Callable):
     import torch._dynamo
     if mode is not None and options is not None:
         raise RuntimeError("Either mode or options can be specified, but both can't be specified at the same time.")
+    if torch.backends.mps.is_available():
```

Comment:
Yeah I think we chose early on to avoid having to tell users to `torch.compile(backend="inductor")` - the whole API needs more work, I can take that on next